addons/product/models/product_template.py

```diff
@@ -774,7 +791,38 @@ def _get_mapped_attribute_names(self, parent_combination=None):
             for attribute_value in all_product_attribute_values
         }
 
-    def _is_combination_possible(self, combination, parent_combination=None):
+    def _is_combination_possible_by_config(self, combination, ignore_no_variant=False):
+        """Return whether the given combination is possible according to the config of attributes on the template
+
+        :param combination: the combination to check for possibility
+        :type combination: recordset `product.template.attribute.value`
+
+        :return: wether the given combination is possible according to the config of attributes on the template
+        :rtype: bool
+        """
+        self.ensure_one()
+
+        valid_pta_line_ids = self.valid_product_template_attribute_line_ids
+
+        if ignore_no_variant:
+            valid_pta_line_ids = self.valid_product_template_attribute_line_ids._without_no_variant_attributes()
```

Comment:
then a bit shorter here too
```suggestion
            attribute_lines = attribute_lines._without_no_variant_attributes()
```